### PR TITLE
4

### PR DESCRIPTION
**Pull request by chrisanicolaou**

### CODE REVIEW

File: controllers/articles.controller.js [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
   updateArticleById,
+  fetchCommentCountByArticleId,
 } = require("../models/articles.model");
 
 exports.getArticleById = async (req, res, next) => {
   try {
     const { article_id } = req.params;
     const result = await fetchArticleById(article_id);
+    const commentCount = await fetchCommentCountByArticleId(article_id);
+    result.comment_count = commentCount;
```
Comment (Rosem10) [inline controllers/articles.controller.js:12]: You've been creative here, and I appreciate that, but we should be getting the comment count back from our db query with the rest of our article result, by joining together our articles and comments tables in our query, please.

**Review by Rosem10 — CHANGES_REQUESTED** [review]
Bit of a rethink on how we're getting that comment count, please!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -31,8 +31,8 @@ describe("GET: /api/topics", () => {
   });
 });
 
-describe("GET: /api/articles/:article_id", () => {
-  test("200: Given a correct file path and existing article_id, returns an article object with the correct properties", () => {
+describe("GET: /api/articles/:article_id (REFACTORED: comment_count)", () => {
```
Comment (dentednerd) [inline __tests__/app.test.js:34]: You should probably leave out notes about refactoring from test descriptions. Maybe pop it into a comment instead?

File: db/connection.js [hunk 3]
```diff
@@ -5,8 +5,18 @@ require("dotenv").config({
   path: `${__dirname}/../.env.${ENV}`,
 });
 
-if (!process.env.PGDATABASE) {
+if (!process.env.PGDATABASE && !process.env.DATABASE_URL) {
   throw new Error("PGDATABASE not set");
```
Comment (dentednerd) [inline db/connection.js:9]: I'm being so super picky here... you'll want to update this string too, to reflect both PGDATABASE and DATABASE_URL! 😄

**Review by dentednerd — APPROVED** [review]
Cracking work Christos! Couple of things to think about, but I won't need to see this again before you merge it.

**Pull request merged**